pandas-stubs/core/groupby/generic.pyi

```diff
@@ -141,6 +141,7 @@ class SeriesGroupBy(GroupBy, Generic[S1, ByT]):
     def idxmax(self, axis: Axis = ..., skipna: bool = ...) -> Series: ...
     def idxmin(self, axis: Axis = ..., skipna: bool = ...) -> Series: ...
     def __iter__(self) -> Iterator[tuple[ByT, Series[S1]]]: ...
+    def diff(self, periods: int = ..., axis: Axis = ...) -> Series: ...
```

Comment:
Could return `Series[S1]` but that would be incorrect for timestamps (and maybe more types).

Reply:
In `series.pyi`, we have that doing subtraction on `TimestampSeries` yields `TimedeltaSeries`, and subtraction on `PeriodSeries` yields `OffsetSeries` .  

So it might be the case that you could do overloads here on `self: SeriesGroupBy[Timestamp, ByT]` to make that work, but it might be tricky to get it to work with `mypy` without creating the other kinds of `Series` like we do in `series.pyi`.

We can leave that as a new issue.  Do you want to do that?